Add autocomplete behaviour for select

kononenko-a-m opened the pull request.

mseemann commented:
thx! will have a look - but I'm afraid I need some time for this :(

mseemann commented on src/components/select/select.spec.ts:
```diff
@@ -476,6 +542,24 @@ describe('MdlSelect', () => {
 @Component({
     selector: 'test-single-component',
     template: `
+        <mdl-select label="{{label}}" floating-label [autoselect]="false" [(ngModel)]="selectedValue">
```
don't you think it would be better to set autoselect to true if you want this feature and set it to false or leave the attribute autoselect away if you don't want the autoselect feature? For me this would be a more natural way how to use additional features...

kononenko-a-m replied:
yeah, here a little bit confused naming. cause actually there is feature which might be named as autoselect. I mean when you press any key with character, select going to autoselect it. and what people means under autoselection (maybe proper name autocomplete actually), it's typing some word which filter out selection, and not actually makes autoselection.

so according to it I decided to call original (existing) behavior as autoselect, which can be disabled, so we can use other type of autoselection.

mseemann commented on src/e2e-app/app/select/select.component.ts:
```diff
@@ -89,10 +92,16 @@ export class SelectDemo {
       personId: this.personId
     });
 
+    this.onAutoselectChange = function (value: string) {
```
why did you not provide a regular function like:

 ```
 public onAutoselectChange = (value: string) => {
    this.autoselectCountries = this.countries.filter((country: any) => country.name.toLowerCase().indexOf(value.toLowerCase()) !== -1);
  };

```

kononenko-a-m replied:
no real reason for it, just used to it, can be replaced to arrow function

mseemann commented:
can you check my comments? also: if there is not enough space under the select component the options will open above the component. But this will hide the select input control I think we have to make sure that the options list is always under the select component if the autoselect feature is used.

kononenko-a-m commented:
about space above select component - we might optionally show it exactly above input so it wouldn't cover input

mseemann commented:
need to see how this looks if the popup opens above the input control...

kononenko-a-m commented:
@mseemann I've renamed feature, so now it's can be enabled, and disabled by default. fixed on mobile, right now it would be shown above input, but it works in such way only for new feature

mseemann commented:
@kononenko-a-m sorry for the delay:( thx for your work!
